### PR TITLE
[pkg/telemetrygen] add support exporting log data

### LINKED ISSUE #12927: telemetrygen subcommand to generate OTLP log events
**Is your feature request related to a problem? Please describe.**

Easily running integration tests for [Vector's](https://github.com/vectordotdev/vector) OTLP logs.

**Describe the solution you'd like**

The WIP `telemetrygen` currently has documented (WIP) support for traces and metrics, we should track in the implementation of logs as well.

**Describe alternatives you've considered**

Today I've just run the opentelemetry-collector to push logs with [otlpexporter](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlpexporter) and [otlphttpexporter](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlphttpexporter).

### PR DESCRIPTION
**Pull request by fatsheep9146**
**Description:** 
support telemetrygen tool to export log data. 

some notes:
- since opentelemetry-go does not support log type, so I have to build the test log data from `pdata` package,
- and export the data to server by directly using gRPC.Client

let me know this is not the best suitable way to implement this.

**Link to tracking Issue:** <Issue number if applicable>
fix #12927

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by fatsheep9146** [comment]
could you help review about this? @mx-psi

**Comment by atoulme** [comment]
Approach looks good, need a worker_test.go to go with this :)

**Review by mx-psi — COMMENTED** [review]
The approach makes sense to me too :) Left a comment on the changelog note and agree with the need of tests

**Comment by fatsheep9146** [comment]
> The approach makes sense to me too :) Left a comment on the changelog note and agree with the need of tests

Done

**Comment by fatsheep9146** [comment]
failed tests are not related to this pr

File: cmd/telemetrygen/internal/logs/logs.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+// Start starts the log telemetry generator
+func Start(cfg *Config) error {
+	logger, err := common.CreateLogger()
+	if err != nil {
+		return err
+	}
+
+	if cfg.UseHTTP {
+		return fmt.Errorf("http is now not supported in log mode of telemetrygen")
```
Comment (mx-psi) [inline cmd/telemetrygen/internal/logs/logs.go:60]: Maybe open an issue for adding support for this?
```suggestion
		return fmt.Errorf("http is not supported by 'telemetrygen logs'")
```

File: cmd/telemetrygen/internal/logs/logs.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	}
+
+	if !cfg.Insecure {
+		return fmt.Errorf("the log mode of telemetrygen only supports insecure grpc")
```
Comment (mx-psi) [inline cmd/telemetrygen/internal/logs/logs.go:64]: Open another issue for adding support for this?
```suggestion
		return fmt.Errorf("'telemetrygen logs' only supports insecure gRPC")
```

**Review by mx-psi — APPROVED** [review]

File: cmd/telemetrygen/internal/logs/logs.go (hunk 1 above)
Comment (fatsheep9146) [reply cmd/telemetrygen/internal/logs/logs.go]: here is the issue #18867

File: cmd/telemetrygen/internal/logs/logs.go (hunk 2 above)
Comment (fatsheep9146) [reply cmd/telemetrygen/internal/logs/logs.go]: here is the issue #18868

**Review by mx-psi — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
The approach looks ok, is the plan to eventually move over to using the otel go sdk to emit logs (to align with how this is done for other signals)? If so we should probably have an issue to track this.

Please rebase :)

**Comment by atoulme** [comment]
@codeboten from experience with metrics and the Go SDK, it might not be possible to use the Go SDK to export things with the fine control required of the tool. For example, with metrics there was no way to fire metrics export on a time-basis that reflects well the arguments passed in, and there was default aggregation/latest value computations happening. So in any case, I don't think we need to be looking to move to the Go SDK in the short term.

**Review by atoulme — APPROVED** [review]

**Comment by fatsheep9146** [comment]
> The approach looks ok, is the plan to eventually move over to using the otel go sdk to emit logs (to align with how this is done for other signals)? If so we should probably have an issue to track this.
> 
> Please rebase :)

@codeboten @atoulme 

I think both of you said make senses, I created a issue #18902 to track this, let's see whether this issue can be resolved when go sdk is ready.

**Pull request merged**